src/files/DCDFile.cpp

```
@@ -0,0 +1,457 @@
+      try
+      {
+         fileStream.open(path, ios::app | ios::binary);
+      }
+      catch (ofstream::failure& e)
+      {
+         cerr << "Exception creating file '" << path << "' : " << endl;
+         cerr << "Couldn't create the file." << endl;
+      }
+
+   }
+}
+
+void DCDFile::readHeader()
+{
+   unsigned int fortcheck1, fortcheck2;
+   long initialPos = fileStream.tellg();
+   //This is the trick for reading binary data from fortran file, see the method checkFortranIOerror for more details.
+   fileStream.read((char*)&fortcheck1, sizeof(unsigned int));               //consistency check 1
+   fileStream.read((char*)HDR, sizeof(char) * 4);                       //first data block written by fortran  : a character array of length 4.
+   fileStream.read((char*)ICNTRL, sizeof(int) * 20);                    //second data block written by fortran : an integer(4) array of length 20.
+   fileStream.read((char*)&fortcheck2, sizeof(unsigned int));         //consistency check 2
+
+   // if the 2 unsigned ints have a different value there was an error
+   if (!checkFortranIOerror(__FILE__, __LINE__, fortcheck1, fortcheck2))
+   {
+      return;
+   }
+
+   HDR[4] = '\0';
+
+   if (ICNTRL[10])
+   {
+      pbc[0] = pbc[1] = pbc[2] = pbc[3] = pbc[4] = pbc[5] = 0.0;
+   }
+    
+   /* Several "lines" of title of length 80 are written to the dcd file by CHARMM */
+   fileStream.read((char*)&fortcheck1, sizeof(unsigned int));
+   fileStream.read((char*)&NTITLE, sizeof(int));
+   if (NTITLE == 0)
+   {
+      TITLE = new char[80 + 1];
+      TITLE[0] = '\0';
+   }
+   else
+   {
+      TITLE = new char[NTITLE * 80 + 1];
+      for (int it = 0; it < NTITLE; it++)
+      {
+         fileStream.read((char*)&TITLE[it * 80], sizeof(char) * 80);
+      }
+      TITLE[NTITLE * 80] = '\0';
+   }
+   fileStream.read((char*)&fortcheck2, sizeof(unsigned int));
+   if (!checkFortranIOerror(__FILE__, __LINE__, fortcheck1, fortcheck2))
+   {
+      return;
```

This should be an exception instead of silently failing